ts/Series/Pie/PieDataLabel.ts

```diff
@@ -653,9 +653,11 @@ namespace ColumnDataLabel {
             if (newSize < center[2]) {
                 center[2] = newSize;
                 center[3] = Math.min( // #3632
-                    relativeLength(options.innerSize || 0, newSize),
+                    options.thickness ?
+                        Math.max(0, newSize - options.thickness * 2) :
```

Comment:
Do we need this `Math.max` here? It doesn't allow `thickness` to be higher than half `size`. But is there a reason why it shouldn't be higher than half `size`? It's working fine with the same use-case of a negative `innerSize` here (no `Math.max` here):
```js
relativeLength(options.innerSize || 0, newSize),
```

Or maybe we do want it to limit both `thickness` **and** `innerSize` based on this [ticket](https://github.com/highcharts/highcharts/issues/3632)?
It seems like the above bug hasn't been fixed for a use case when the dataLabels are enabled.
Demo: http://jsfiddle.net/BlackLabel/1ys0bkz5/

Reply:
`thickness` should be limited so that `innerSize` is not set to more than `size` and is not less than 0.